httpclient/src/test/kotlin/tbdex/sdk/httpclient/E2ETest.kt

```diff
@@ -0,0 +1,233 @@
+        "cHVibGljS2V5SndrIjp7ImNydiI6IkVkMjU1MTkiLCJrdHkiOiJPS1AiL" +
+        "CJ4IjoieEVnVkNvVEYtUHpXejQ2elFuUEVQS2RtTk9mTWpuVXVhWWx0Qll" +
+        "UZGxtWSJ9LCJwdXJwb3NlcyI6WyJhdXRoZW50aWNhdGlvbiIsImFzc2Vyd" +
+        "Glvbk1ldGhvZCJdLCJ0eXBlIjoiSnNvbldlYktleTIwMjAifV0sInNlcnZ" +
+        "pY2VzIjpbeyJpZCI6InBmaSIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHA6L" +
+        "y9sb2NhbGhvc3Q6OTAwMCIsInR5cGUiOiJQRkkifV19fV0sInVwZGF0ZUNv" +
+        "bW1pdG1lbnQiOiJFaUR0YnphSndFaG5kbS05Y1Q0eDc2OGFTUmIxS29nRVh" +
+        "0MzJEbkZ5OU05Q2xnIn0sInN1ZmZpeERhdGEiOnsiZGVsdGFIYXNoIjoiRWl" +
+        "Ed3l2OXhsQW83cUt4NjRjOThHeTJKMFliTldwbGw2ZkNfcmJ5eGZ3b29pQSIs" +
+        "InJlY292ZXJ5Q29tbWl0bWVudCI6IkVpQ1hMaWVpRnlCa1hKYU84eEVwOTV0Z" +
+        "Ed3bUlaVG4wZEV4UnpBR3MyaXJueHcifX0"
+    val myDid = DidKey.create(InMemoryKeyManager())
+
+    println("let's do a tbdex transaction!")
+    println("Getting offerings...")
+    val getOfferingsResponse = client.getOfferings(pfiDid, GetOfferingsFilter("BTC", "KES"))
+
+    if (getOfferingsResponse is ErrorResponse) {
+      throw AssertionError(
+        "Error getting offerings. " +
+          "Errors: ${getOfferingsResponse.errors?.joinToString(", ") { it.detail }}"
+      )
+    }
+
+    val offerings = (getOfferingsResponse as GetOfferingsResponse).data
+    println(
+      "Got offerings! payin ${(offerings.first() as Offering).data.payinCurrency.currencyCode}, " +
+        "payout ${(offerings.first() as Offering).data.payoutCurrency.currencyCode}"
+    )
+
+    if (offerings.isEmpty()) {
+      throw AssertionError("Offerings are empty")
+    }
+
+    val firstOfferingId = offerings[0].metadata.id
+    val vc = VerifiableCredential.create(
+      type = "SanctionCredential",
+      issuer = myDid.uri,
+      subject = myDid.uri,
+      data = mapOf("fullName" to "Lainey Wilson", "country" to "US")
+    )
+
+    val vcJwt = vc.sign(myDid)
+    val rfqData = RfqData(
+      offeringId = firstOfferingId,
+      payinSubunits = "100",
+      payinMethod = SelectedPaymentMethod(
+        kind = "NGN_ADDRESS",
+        paymentDetails = mapOf("walletAddress" to "ngn-wallet-address")
+      ),
+      payoutMethod = SelectedPaymentMethod(
+        kind = "BANK_Access Bank",
+        paymentDetails = mapOf(
+          "accountNumber" to "1234567890",
+          "reason" to "gift",
+          "accountHolderName" to "lainey wilson"
+        )
+      ),
+      claims = listOf(vcJwt)
+    )
+    val rfq = Rfq.create(pfiDid, myDid.uri, rfqData)
+    rfq.sign(myDid)
+    println("Sending RFQ against first offering id: ${offerings[0].metadata.id}")
+    val sendRfqResponse = client.sendMessage(rfq)
+
+    if (sendRfqResponse is ErrorResponse) {
+      throw AssertionError(
+        "Error returned from sending RFQ. " +
+          "Errors: ${sendRfqResponse.errors?.joinToString(", ") { it.detail }}"
+      )
+    }
+    println("SendRfqResponse is success?? ${sendRfqResponse is SendMessageResponse}")
```

Comment:
Should this return if sending RFQ failed?

Reply:
the few lines above this one checks if SendRfqResponse is ErrorResponse then throws AssertionError

```if (sendRfqResponse is ErrorResponse) {
      throw AssertionError(
        "Error returned from sending RFQ. " +
          "Errors: ${sendRfqResponse.errors?.joinToString(", ") { it.detail }}"
      )
    }```

Reply:
i could remove these they're sort of implicit. i just wanted extra assurances logged at the time of testing